tools/toolchains/gcc.py

```diff
@@ -82,6 +86,17 @@ def __init__(self, target,  notify=None, macros=None,
 
         self.flags["common"] += self.cpu
 
+        # Coverage is turned On if coverage_filter is not empty. This means all sources are compiled with coverage
+        # macro and application is linked with coverage flags.
+        # Only source files that match regex from coverage_filter list are compiled with coverage flags. Since turning
+        # On code coverage on all sources increases static data size it overlaps with stack allocation and causes linker
+        # error.
+        self.coverage_filter = coverage_filter
+        if self.coverage_filter:
+            self.flags["common"].append("-g")
+            self.flags["common"].append("-O0")
```

Comment:
This may be a problem. But see below for every source file all the macros are repeated. Hence there must be another issue:

```
[OS ERROR] Command: C:\Program Files (x86)\GNU Tools ARM Embedded\4.9 2015q3\bin\arm-none-eabi-g++ -std=gnu++98 -fno-rtti -Wvla -c -Wall -Wextra -Wno-unused-parameter -Wno-missing-fiel
d-initializers -fmessage-length=0 -fno-exceptions -fno-builtin -ffunction-sections -fdata-sections -funsigned-char -MMD -fno-delete-null-pointer-checks -fomit-frame-pointer -Os -mcpu=c
ortex-m4 -mthumb -mfpu=fpv4-sp-d16 -mfloat-abi=softfp -DFEATURE_LWIP=1 -D__MBED__=1 -DDEVICE_I2CSLAVE=1 -D__FPU_PRESENT=1 -DTARGET_Freescale -DDEVICE_PORTINOUT=1 -DTARGET_RTOS_M4_M7 -D
DEVICE_LOWPOWERTIMER=1 -DDEVICE_RTC=1 -DTOOLCHAIN_object -D__CMSIS_RTOS -DFSL_RTOS_MBED -DDEVICE_STORAGE=1 -DTARGET_KPSDK_MCUS -DTOOLCHAIN_GCC -DDEVICE_SPISLAVE=1 -DTARGET_CORTEX_M -DT
ARGET_KSDK2_MCUS -DTARGET_LIKE_CORTEX_M4 -DDEVICE_ANALOGOUT=1 -DTARGET_M4 -DTARGET_UVISOR_UNSUPPORTED -DTARGET_K64F -DDEVICE_SERIAL=1 -DDEVICE_INTERRUPTIN=1 -DDEVICE_I2C=1 -DDEVICE_POR
TOUT=1 -D__CORTEX_M4 -DDEVICE_STDIO_MESSAGES=1 -DCPU_MK64FN1M0VMD12 -DTARGET_LIKE_MBED -DTARGET_FF_ARDUINO -DTARGET_KPSDK_CODE -DTARGET_RELEASE -DDEVICE_SERIAL_FC=1 -DFEATURE_STORAGE=1
 -DDEVICE_TRNG=1 -D__MBED_CMSIS_RTOS_CM -DDEVICE_SLEEP=1 -DTOOLCHAIN_GCC_ARM -DTARGET_FRDM -DDEVICE_SPI=1 -DDEVICE_ERROR_RED=1 -DMBED_BUILD_TIMESTAMP=1478097168.76 -DDEVICE_ANALOGIN=1
-DDEVICE_PWMOUT=1 -DDEVICE_PORTIN=1 -DTARGET_MCU_K64F -DARM_MATH_CM4 @C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\.includes_2ee9662607fcf9800041536b8
3d29577.txt -include C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\mbed_config.h -MD -MF C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\te
sts\K64F\GCC_ARM\drivers\AnalogIn.d -o C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\drivers\AnalogIn.o C:\Users\azikha01\Documents\GitHub\mazimkhan\mb
ed-os\drivers\AnalogIn.cpp -DFEATURE_LWIP=1 -D__MBED__=1 -DDEVICE_I2CSLAVE=1 -D__FPU_PRESENT=1 -DTARGET_Freescale -DDEVICE_PORTINOUT=1 -DTARGET_RTOS_M4_M7 -DDEVICE_LOWPOWERTIMER=1 -DDE
VICE_RTC=1 -DTOOLCHAIN_object -D__CMSIS_RTOS -DFSL_RTOS_MBED -DDEVICE_STORAGE=1 -DTARGET_KPSDK_MCUS -DTOOLCHAIN_GCC -DDEVICE_SPISLAVE=1 -DTARGET_CORTEX_M -DTARGET_KSDK2_MCUS -DTARGET_L
IKE_CORTEX_M4 -DDEVICE_ANALOGOUT=1 -DTARGET_M4 -DTARGET_UVISOR_UNSUPPORTED -DTARGET_K64F -DDEVICE_SERIAL=1 -DDEVICE_INTERRUPTIN=1 -DDEVICE_I2C=1 -DDEVICE_PORTOUT=1 -D__CORTEX_M4 -DDEVI
CE_STDIO_MESSAGES=1 -DCPU_MK64FN1M0VMD12 -DTARGET_LIKE_MBED -DTARGET_FF_ARDUINO -DTARGET_KPSDK_CODE -DTARGET_RELEASE -DDEVICE_SERIAL_FC=1 -DFEATURE_STORAGE=1 -DDEVICE_TRNG=1 -D__MBED_C
MSIS_RTOS_CM -DDEVICE_SLEEP=1 -DTOOLCHAIN_GCC_ARM -DTARGET_FRDM -DDEVICE_SPI=1 -DDEVICE_ERROR_RED=1 -DMBED_BUILD_TIMESTAMP=1478097168.76 -DDEVICE_ANALOGIN=1 -DDEVICE_PWMOUT=1 -DDEVICE_
PORTIN=1 -DTARGET_MCU_K64F -DARM_MATH_CM4 @C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\.includes_2ee9662607fcf9800041536b83d29577.txt -include C:\Use
rs\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\mbed_config.h -MD -MF C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\drivers\Bu
sIn.d -o C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\drivers\BusIn.o C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\drivers\BusIn.cpp -DFEATURE
_LWIP=1 -D__MBED__=1 -DDEVICE_I2CSLAVE=1 -D__FPU_PRESENT=1 -DTARGET_Freescale -DDEVICE_PORTINOUT=1 -DTARGET_RTOS_M4_M7 -DDEVICE_LOWPOWERTIMER=1 -DDEVICE_RTC=1 -DTOOLCHAIN_object -D__CM
SIS_RTOS -DFSL_RTOS_MBED -DDEVICE_STORAGE=1 -DTARGET_KPSDK_MCUS -DTOOLCHAIN_GCC -DDEVICE_SPISLAVE=1 -DTARGET_CORTEX_M -DTARGET_KSDK2_MCUS -DTARGET_LIKE_CORTEX_M4 -DDEVICE_ANALOGOUT=1 -
DTARGET_M4 -DTARGET_UVISOR_UNSUPPORTED -DTARGET_K64F -DDEVICE_SERIAL=1 -DDEVICE_INTERRUPTIN=1 -DDEVICE_I2C=1 -DDEVICE_PORTOUT=1 -D__CORTEX_M4 -DDEVICE_STDIO_MESSAGES=1 -DCPU_MK64FN1M0V
MD12 -DTARGET_LIKE_MBED -DTARGET_FF_ARDUINO -DTARGET_KPSDK_CODE -DTARGET_RELEASE -DDEVICE_SERIAL_FC=1 -DFEATURE_STORAGE=1 -DDEVICE_TRNG=1 -D__MBED_CMSIS_RTOS_CM -DDEVICE_SLEEP=1 -DTOOL
CHAIN_GCC_ARM -DTARGET_FRDM -DDEVICE_SPI=1 -DDEVICE_ERROR_RED=1 -DMBED_BUILD_TIMESTAMP=1478097168.76 -DDEVICE_ANALOGIN=1 -DDEVICE_PWMOUT=1 -DDEVICE_PORTIN=1 -DTARGET_MCU_K64F -DARM_MAT
H_CM4 @C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\.includes_2ee9662607fcf9800041536b83d29577.txt -include C:\Users\azikha01\Documents\GitHub\mazimkh
an\mbed-os\.build\tests\K64F\GCC_ARM\mbed_config.h -MD -MF C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\drivers\BusInOut.d -o C:\Users\azikha01\Docume
nts\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\drivers\BusInOut.o C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\drivers\BusInOut.cpp -DFEATURE_LWIP=1 -D__MBED__=1 -DDEVI
CE_I2CSLAVE=1 -D__FPU_PRESENT=1 -DTARGET_Freescale -DDEVICE_PORTINOUT=1 -DTARGET_RTOS_M4_M7 -DDEVICE_LOWPOWERTIMER=1 -DDEVICE_RTC=1 -DTOOLCHAIN_object -D__CMSIS_RTOS -DFSL_RTOS_MBED -D
DEVICE_STORAGE=1 -DTARGET_KPSDK_MCUS -DTOOLCHAIN_GCC -DDEVICE_SPISLAVE=1 -DTARGET_CORTEX_M -DTARGET_KSDK2_MCUS -DTARGET_LIKE_CORTEX_M4 -DDEVICE_ANALOGOUT=1 -DTARGET_M4 -DTARGET_UVISOR_
UNSUPPORTED -DTARGET_K64F -DDEVICE_SERIAL=1 -DDEVICE_INTERRUPTIN=1 -DDEVICE_I2C=1 -DDEVICE_PORTOUT=1 -D__CORTEX_M4 -DDEVICE_STDIO_MESSAGES=1 -DCPU_MK64FN1M0VMD12 -DTARGET_LIKE_MBED -DT
ARGET_FF_ARDUINO -DTARGET_KPSDK_CODE -DTARGET_RELEASE -DDEVICE_SERIAL_FC=1 -DFEATURE_STORAGE=1 -DDEVICE_TRNG=1 -D__MBED_CMSIS_RTOS_CM -DDEVICE_SLEEP=1 -DTOOLCHAIN_GCC_ARM -DTARGET_FRDM
 -DDEVICE_SPI=1 -DDEVICE_ERROR_RED=1 -DMBED_BUILD_TIMESTAMP=1478097168.76 -DDEVICE_ANALOGIN=1 -DDEVICE_PWMOUT=1 -DDEVICE_PORTIN=1 -DTARGET_MCU_K64F -DARM_MATH_CM4 @C:\Users\azikha01\Do
cuments\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\.includes_2ee9662607fcf9800041536b83d29577.txt -include C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64
F\GCC_ARM\mbed_config.h -MD -MF C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\drivers\BusOut.d -o C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\
.build\tests\K64F\GCC_ARM\drivers\BusOut.o C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\drivers\BusOut.cpp -DFEATURE_LWIP=1 -D__MBED__=1 -DDEVICE_I2CSLAVE=1 -D__FPU_PRESENT=1 -
DTARGET_Freescale -DDEVICE_PORTINOUT=1 -DTARGET_RTOS_M4_M7 -DDEVICE_LOWPOWERTIMER=1 -DDEVICE_RTC=1 -DTOOLCHAIN_object -D__CMSIS_RTOS -DFSL_RTOS_MBED -DDEVICE_STORAGE=1 -DTARGET_KPSDK_M
CUS -DTOOLCHAIN_GCC -DDEVICE_SPISLAVE=1 -DTARGET_CORTEX_M -DTARGET_KSDK2_MCUS -DTARGET_LIKE_CORTEX_M4 -DDEVICE_ANALOGOUT=1 -DTARGET_M4 -DTARGET_UVISOR_UNSUPPORTED -DTARGET_K64F -DDEVIC
E_SERIAL=1 -DDEVICE_INTERRUPTIN=1 -DDEVICE_I2C=1 -DDEVICE_PORTOUT=1 -D__CORTEX_M4 -DDEVICE_STDIO_MESSAGES=1 -DCPU_MK64FN1M0VMD12 -DTARGET_LIKE_MBED -DTARGET_FF_ARDUINO -DTARGET_KPSDK_C
ODE -DTARGET_RELEASE -DDEVICE_SERIAL_FC=1 -DFEATURE_STORAGE=1 -DDEVICE_TRNG=1 -D__MBED_CMSIS_RTOS_CM -DDEVICE_SLEEP=1 -DTOOLCHAIN_GCC_ARM -DTARGET_FRDM -DDEVICE_SPI=1 -DDEVICE_ERROR_RE
D=1 -DMBED_BUILD_TIMESTAMP=1478097168.76 -DDEVICE_ANALOGIN=1 -DDEVICE_PWMOUT=1 -DDEVICE_PORTIN=1 -DTARGET_MCU_K64F -DARM_MATH_CM4 @C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\
.build\tests\K64F\GCC_ARM\.includes_2ee9662607fcf9800041536b83d29577.txt -include C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\mbed_config.h -MD -MF C
:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\drivers\CAN.d -o C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\.build\tests\K64F\GCC_ARM\drivers\CA
N.o C:\Users\azikha01\Documents\GitHub\mazimkhan\mbed-os\drivers\CAN.cpp -DFEATURE_LWIP=1 -D__MBED__=1 -DDEVICE_I2CSLAVE=1 -D__FPU_PRESENT=1 -DTARGET_Freescale -DDEVICE_PORTINOUT=1 -DT
ARGET_RTOS_M4_M7 -DDEVICE_LOWPOWERTIMER=1 -DDEVICE_RTC=1 -DTOOLCHAIN_object -D__CMSIS_RTOS -DFSL_RTOS_MBED -DDEVICE_STORAGE=1 -DTARGET_KPSDK_MCUS -DTOOLCHAIN_GCC -DDEVICE_SPISLAVE=1 -D
TARGET_CORTEX_M -DTARGET_KSDK2_MCUS -DTARGET_LIKE_CORTEX_M4 -DDEVICE_ANALOGOUT=1 -DTARGET_M4 -DTARGET_UVISOR_UNSUPPORTED -DTARGET_K64F -DDEVICE_SERIAL=1 -DDEVICE_INTERRUPTIN=1 -DDEVICE
_I2C=1 -DDEVICE_PORTOUT=1 -D__CORTEX_M4 -DDEVICE_STDIO_MESSAGES=1 -DCPU_MK64FN1M0VMD12 -DTARGET_LIKE_MBED -DTARGET_FF_ARDUINO -DTARGET_KPSDK_CODE -DTARGET_RELEASE -DDEVICE_SERIAL_FC=1
```